packages/endpoints-organizations-registrar/src/entities/registrar-consents/repos/registrar-consent-repo-extension.js

```diff
@@ -14,26 +14,31 @@
  * limitations under the License.
  */
 
+const { castArray, map } = require('lodash/fp');
 const { ObjectId } = require('mongodb');
 const { nanoid } = require('nanoid');
 
 const registrarConsentRepoExtension = (parent) => ({
-  registerConsent: (
-    { userId, organization, type, version, ...rest },
-    ...args
-  ) => {
-    return parent.insert(
-      {
+  registerConsent: (consentParams, ...args) => {
+    let consents = castArray(consentParams);
+    consents = map((consent) => {
+      const { userId, organization, type, version, ...rest } = consent;
+      return {
         consentId: nanoid(),
         userId,
-        organizationId: new ObjectId(organization._id),
+        organizationId: organization
+          ? new ObjectId(organization._id)
```

Comment:
1. why is `new ObjectId` needed?
2. `createdAt` should be omitted

Reply:
> why is new ObjectId needed?

done in [80bc7e6](https://github.com/velocitycareerlabs/velocitycore/pull/34/commits/80bc7e6e86810002ee32181059218770d9650130)

> createdAt should be omitted

don't we want a timestamp for the consent?

> I think the argument should be organizationId to make it consistent with serviceId

done in [e85f64a](https://github.com/velocitycareerlabs/velocitycore/pull/34/commits/e85f64ab3f414e2c28a6d01fd0d8579414142ec5)

Reply:
> don't we want a timestamp for the consent?

it should be automatic

Reply:
done, [8d1d3dd](https://github.com/velocitycareerlabs/velocitycore/pull/34/commits/8d1d3ddb2fee2768cbf0d214f3f0939db1b4d3a4)